src/common/structures/api.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { model } from "amf-client-js";
+
+import { resolveApiModel, parseRamlFile, getAllDataTypes } from "../parser";
+
+import { Name } from "./name";
+import { ApiTree } from "./apiTree";
+import { RestApi } from "../../download/exchangeTypes";
+import path from "path";
+import fs from "fs-extra";
+
+/**
+ * An API represented as an AMF model. Includes the extracted data types
+ * defined in the spec. Common transformations of the name are cached for
+ * reference in templates and for use in assembling file paths.
+ */
+export class Api extends ApiTree {
```

Comment:
Not 100% convinced about these names. Given `Api` and `ApiTree`, I would assume that `Api` is a base class, and `ApiTree` is some kind of structure containing the relationships between `Api`s.

Reply:
Definitely can discuss it.  I might change the names in my followup PR